### ISSUE TITLE
"Bad Gateway" error when using devpi push to push from index to PyPI

### ISSUE THREAD

**Issue by nicoddemus** [devpi/devpi#449]
Hi,

I'm trying to push a package from an index to PyPI but I'm getting this response:

```
current devpi index: https://devpi.net/nicoddemus/dev (logged in as nicoddemus)
~/.pydistutils.cfg     : no config file exists
~/.pip/pip.conf        : no config file exists
~/.buildout/default.cfg: no config file exists
always-set-cfg: no
using pypirc /home/vagrant/.pypirc
PUSH https://devpi.net/nicoddemus/dev
502 Bad Gateway: {
  "type": "actionlog",
  "result": [
    [
      410,
      "register",
      "pytest",
      "3.2.1"
    ]
  ]
}
```

The command line I used is:

```
$ devpi push pytest==3.2.1 pypi:pypi
```

And my `~/.pypirc` reads:

```
[pypi]
username:nicoddemus
password:XXXXXXXXXXXX
repository:https://upload.pypi.io/legacy/
```

I tried to remove the `repository` entry from my `~/pypirc` but still get the same error (as indicated by https://packaging.python.org/guides/migrating-to-pypi-org/).

Here's my devpi version:

```
$ devpi --version
devpi-client 3.0.0

current devpi server: https://devpi.net/
    devpi-findlinks 2.0.0
    devpi-server 4.2.0
    devpi-web 3.1.1
```

Any advice is greatly appreciated.

**Comment by fschulze** [comment]
The reply from PyPI changed slightly. I have a fix and hope to release it soon. 

Regards,
Florian Schulze

**Comment by fschulze** [comment]
Fixed since 4.3.1rc1.
